### PR TITLE
feature/extract multiple numbers from text

### PR DESCRIPTION
**Pull request by JarbasAl**
## Description

extract_number assumes only 1 number is present in text, this adds a extract_numbers method to support cases like "test one two three"

normalize only accounts for numbers up to 20 (word to digit), method is improved to handle all extracted numbers by above method (needs improvements in further PRs)

## How to test

check unit tests

## Contributor license agreement signed?
CLA [ yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/util/lang/parse_en.py [hunk 1]
```diff
@@ -1039,15 +1040,21 @@ def normalize_en(text, remove_articles):
...
-                       "thirteen", "fourteen", "fifteen", "sixteen",
-                       "seventeen", "eighteen", "nineteen", "twenty"]
-
-        if word in textNumbers:
-            word = str(textNumbers.index(word))
-
         normalized += " " + word
 
+    # needs to be here to avoid import error
+    from mycroft.util.parse import extract_numbers
```
Comment (forslund) [inline mycroft/util/lang/parse_en.py]: I'm not a fan of doing these kinds of workarounds for cyclic imports.

What do you think of creating a base method for extract_numbers in `lang/parse_common.py` which takes the extract_number function as an argument and a just a thin wrapper function in `util/parse.py`?

Comment (JarbasAl) [reply mycroft/util/lang/parse_en.py]: sounds good!

**Comment by JarbasAl** [comment]
added support for more languages, but since i don't speak them i would appreciate someone to write unittests

it is using the "extract number" logic already implemented for each language recursively, so it should be correct

**Comment by forslund** [comment]
Is the intent to only match numbers directly following eachother?

"so two beers for two bears" should return `[2]` and not `[2, 2]`?

**Comment by JarbasAl** [comment]
it should extract every number, "1 dog, seven pigs, macdonald had a farm, 3 times 5 macarena" should also work

**Comment by forslund** [comment]
The example I gave above didn't work for me.

**Comment by JarbasAl** [comment]
you are right, i need to account for duplicate numbers, what is happening is that when it find the first "two" it replaces all instances in the string instead of only the consumed one

will account for this

**Comment by JarbasAl** [comment]
moved main logic into parse_common to make it easier for supporting other languages

improved normalization of numbers, now "two twenty two" correctly becomes "2 22" instead of "2 twenty 2"

**Comment by JarbasAl** [comment]
geez, how come pycharms auto format is creating new pep8 issues instead of solving them...

**Comment by forslund** [comment]
Looks good. I can't figure out how to clean up the commit history so I'll just squash it all into one commit and merge it.

**Comment by forslund** [comment]
Ok a conflict seem to have occurred when I merged the datetime update. Can you look at it?

**Comment by JarbasAl** [comment]
rebased and added some more relevant unittests, wanted to leave some TODOs in there to make some limitations clear

follow up PR for pronounce_numbers (ordinals and long scale support) should fix some of those

follow up PR for issue https://github.com/MycroftAI/mycroft-core/issues/1875 will fix remaining edge cases

**Pull request merged**